packages/story-editor/src/app/story/effects/useLoadStory.js

```diff
@@ -56,6 +56,8 @@ function loadStory(storyId, post, restore, clearHistory) {
     terms,
   } = post;
 
+  const date = origDate === modified ? null : origDate;
```

Comment:
@swissspidy, the issue here is, that when I go to create a new story, the already preloaded story data (loaded in from the cached REST response in the document) has a fixed date of "now" rather than no date at all (`null`). This is just an auto-draft created of the new empty story, but it still contains initial data including a publish date.

The above code assumes that only such a brand new story auto-draft has `date` and `modified` set to the same value and assumes the publish date to be `null` in such a case and this works. The application will then save `null` and `null` will correctly be loaded back in once the application reloads.

A better solution here would be for the original auto-draft to simply have a publish date of `null` in the first place, but I'm not sure where to fix that.

If that is not achievable, we might still want to look for a better way to test for "brand new empty story" than the above check. Technically a just published story might have modified and publish dates set to the same timestamp (though that doesn't appear to happen under normal circumstances), and this snippet above would then unset the publish date.

So, my thoughts for the proper fix in priority:

1. Make sure an initial auto-draft of a new empty story has `date` set to `null` rather than current timestamp,
2. If that doesn't work, replace the above check with a foolproof way to determine if a story is brand new,
3. If that is not possible, go with what I have above.

Remember that all of the above should also work for a new story created from a template. My fix above does do that, but we must make sure that any alternative (and better) fix also does that.

Reply:
Thanks for digging into this. Not sure if there are other implications when setting the `date` to `null` in this case. But I gave it a try in #11598 if you wanna test it.

IMHO it does paper over the underlying issue that we send the `date` field when saving a story, even though we haven't touched it (same goes for all other fields of the story). Compare this to Gutenberg where only the actually changed fields are sent to the REST API, not all fields. Which is why it does not suffer this same problem.

This now just works because we set the date to `null` for drafts and then send `'date' => null` when saving a story, which then means it won't get updated in the db.

Reply:
So #11598 only works if also doing it for `draft` and not just `auto-draft`, but that seems to break the list view on the dashboard, which expects proper dates.

Reply:
@swissspidy, what's your recommendation here? I will completely refer to you when it comes to data integrity. In my list above, should we go with option 1, 2, or 3? And if 2, how would you recommend we then go about this?

Reply:
If we do look into your original idea of only sending changed data to the API, I don't see how that fixes the issue, that the client would load a non-null timestamp in and then display the publish date as that timestamp (which is part of the issue). Sure, we wouldn't be re-sending the data to the backend, but we would still be displayed the wrong date in the client, no?

Reply:
@swissspidy, fresh ping on this issue now that you're back.

Reply:
You're right, we might still be displaying the wrong date in the client.

In that case, based on https://github.com/WordPress/gutenberg/blob/9580b45e6e18dd06076af9f7e1ea66babee22bf5/packages/editor/src/store/selectors.js#L756-L786, we should be able to go with (2), like this:

```suggestion
  const date =
    ['draft', 'auto-draft', 'pending'].includes(status) &&
    (origDate === modified || !date)
      ? null
      : origDate;
```

That seems to work fine in my testing right now.

WDYT?

Reply:
Sounds great, let's go with that!

Reply:
Implemented in 93612f128bd656921127367ecace9fa7107bd576.